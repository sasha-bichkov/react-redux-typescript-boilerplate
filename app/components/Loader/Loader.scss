.Loader {
  position: relative;
  display: inline-block;

  width: 80px;
  height: 80px;
}

.Loader div {
  position: absolute;
  display: block;

  box-sizing: border-box;
  width: 64px;
  height: 64px;
  margin: 8px;
  border: 8px solid #000;
  border-color: #000 transparent transparent;
  border-radius: 50%;

  animation: lds-ring 1.2s cubic-bezier(.5, 0, .5, 1) infinite;
}

.Loader div:nth-child(1) {
  animation-delay: -.45s;
}

.Loader div:nth-child(2) {
  animation-delay: -.3s;
}

.Loader div:nth-child(3) {
  animation-delay: -.15s;
}

@keyframes lds-ring {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
