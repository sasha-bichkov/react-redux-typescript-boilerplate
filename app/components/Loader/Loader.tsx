import React from 'react'

import './Loader.scss'

const Loader: React.FunctionComponent = () => {
  return (
    <div className="Loader">
      <div></div>
      <div></div>
      <div></div>
      <div></div>
    </div>
  )
}

export default Loader
