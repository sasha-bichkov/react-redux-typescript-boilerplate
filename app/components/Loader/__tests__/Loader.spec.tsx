import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Loader from '../Loader'

describe('Loader', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Loader />)

    expect(asFragment()).toMatchSnapshot()
  })
})
