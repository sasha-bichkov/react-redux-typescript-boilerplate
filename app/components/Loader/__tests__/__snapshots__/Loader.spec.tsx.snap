// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<div
  className="Loader"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
